symbolic-debuginfo/src/breakpad.rs

```diff
@@ -1273,17 +1273,14 @@ impl<'data> Iterator for BreakpadSymbolIterator<'data> {
     type Item = Symbol<'data>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        while let Some(result) = self.records.next() {
-            if let Ok(record) = result {
-                return Some(Symbol {
-                    name: Some(Cow::Borrowed(record.name)),
-                    address: record.address,
-                    size: 0,
-                });
-            }
-        }
-
-        None
+        self.records
+            .next()
+            .and_then(|result| result.ok())
+            .map(|record| Symbol {
+                name: Some(Cow::Borrowed(record.name)),
+                address: record.address,
+                size: 0,
+            })
```

Comment:
Does this do the same thing as the original version? Originally you called `next` in a loop, here you call it only once and then map over it. Maybe I'm missing something, but I think you want something like
```
self.records.find_map(Result::ok).map(|record| Symbol {
    name: Some(Cow::Borrowed(record.name)),
    address: record.address,
    size: 0
})
```

Reply:
You're right.  I'm kind of sad the tests passed with this though, but hey.

Is it safe to assume find_map advanced the iterator correctly and leaves it in the right state for the next call?

Reply:
I think that's a safe assumption, yes.